File: leetcode/1019-squares-of-a-sorted-array/solution.java
class Solution {
    public int[] sortedSquares(int[] nums) {

        int l = 0, r = nums.length - 1;
        int[] res = new int[nums.length];
        int counter = nums.length - 1;
        while(l <= r){
            if(Math.abs(nums[l]) > Math.abs(nums[r])){
                res[counter--] = nums[l] * nums[l];
                l++;
            }
            else{
                res[counter--] =nums[r] * nums[r];
                r--;
            }
        }
        return res;
    }
}

File: leetcode/1019-squares-of-a-sorted-array/solution.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        l = 0
        r = x = n - 1
        res = [0] * n

        while l <= r:
            if abs(nums[r]) >= abs(nums[l]):
                res[x] = nums[r] ** 2
                r -= 1
            else:
                res[x] = nums[l] ** 2
                l += 1
            x -= 1
        
        return res

